kili/mutations/asset/__init__.py

```diff
@@ -199,36 +210,53 @@ def update_properties_in_assets(self,
-            'priority',
-            'jsonMetadata',
-            'consensusMark',
-            'honeypotMark',
-            'toBeLabeledBy',
-            'content',
-            'jsonContent',
-            'status',
-            'isUsedForConsensus',
-            'isHoneypot'
-        ]
-        to_be_labeled_by_array = data[5]
-        should_reset_to_be_labeled_by_array = list(
-            map(is_none_or_empty, to_be_labeled_by_array))
-        for i, properties in enumerate(zip(*data)):
-            for _property, property_value in zip(property_names, properties):
-                data_array[i][_property] = property_value
-        for i in range(nb_assets_to_modify):
-            data_array[i]['shouldResetToBeLabeledBy'] = should_reset_to_be_labeled_by_array[i]
-        variables = {
-            'whereArray': where_array,
-            'dataArray': data_array
-        }
-        result = self.auth.client.execute(
-            GQL_UPDATE_PROPERTIES_IN_ASSETS, variables)
-        return format_result('data', result, Asset)
+        results = []
+        for batch_number, paginated_properties \
+                in enumerate(batch_iterators_builder([asset_ids]+list_of_properties)):
+            mutation_start = time.time()
+            asset_ids_batch = paginated_properties[0]
+            list_of_properties_batch = paginated_properties[1:]
+            where_array = [{'id': asset_id} for asset_id in asset_ids_batch]
+            nb_assets_to_modify = len(where_array)
+            data_array = [{} for i in range(len(where_array))]
+            data = list(map(partial(convert_to_list_of_none,
+                                    length=nb_assets_to_modify), list_of_properties_batch))
+            property_names = [
+                'externalId',
+                'priority',
+                'jsonMetadata',
+                'consensusMark',
+                'honeypotMark',
+                'toBeLabeledBy',
+                'content',
+                'jsonContent',
+                'status',
+                'isUsedForConsensus',
+                'isHoneypot'
+            ]
+            to_be_labeled_by_array = data[5]
+            should_reset_to_be_labeled_by_array = list(
+                map(is_none_or_empty, to_be_labeled_by_array))
+            for i, properties in enumerate(zip(*data)):
+                for _property, property_value in zip(property_names, properties):
+                    data_array[i][_property] = property_value
+            for i in range(nb_assets_to_modify):
+                data_array[i]['shouldResetToBeLabeledBy'] = should_reset_to_be_labeled_by_array[i]
+            variables = {
+                'whereArray': where_array,
+                'dataArray': data_array
+            }
+            result = self.auth.client.execute(
+                GQL_UPDATE_PROPERTIES_IN_ASSETS, variables)
+            mutation_time = time.time() - mutation_start
+            if mutation_time < THROTTLING_DELAY:
```

Comment:
the batch logic should be abstracted away from the implementation of the mutations, so that we don't nee to rewrite the logic anytime we add a method. It could be a decorator for example

Reply:
Seen with you, I accept as it's improves the user experience, but it should be refactorized and batched in the end